salt/cloud/clouds/nova.py

```diff
@@ -726,12 +726,18 @@ def request_instance(vm_=None, call=None):
 
         else:
             pool = floating_ip_conf.get('pool', 'public')
-            for fl_ip, opts in six.iteritems(conn.floating_ip_list()):
-                if opts['fixed_ip'] is None and opts['pool'] == pool:
-                    floating_ip = fl_ip
-                    break
-            if floating_ip is None:
+            try:
                 floating_ip = conn.floating_ip_create(pool)['ip']
+            except Exception as exc:
+                log.info('A new ip address was unable to be allocated. '
+                         'An ip address will be pulled from the already allocated list, '
+                         'This will cause a race condition when building in parallel.')
+                for fl_ip, opts in six.iteritems(conn.floating_ip_list()):
+                    if opts['fixed_ip'] is None and opts['pool'] == pool:
+                        floating_ip = fl_ip
+                        break
+                if floating_ip is None:
+                    log.error('No ip addresses available to allocate for this server: {0}', vm_['name'])
```

Comment:
sorry, yeah, it should actually be '%s' in the log message, log functions do the substitution for you.

or it could be a .format.